lib/commands/commands/riot.ts

```diff
@@ -32,39 +34,157 @@ export const getSummonerId = async (ign:string, server:string) => {
         return;
-    }
+    })
+
     const matchBaseData = recentGames.data.matches[0];
     const matchId = matchBaseData
         ? matchBaseData.gameId
-        : undefined
-    
-    try {
-        const path = `https://${realm}.api.riotgames.com/lol/match/v4/timelines/by-match/${matchId}?api_key=${config.RIOT_API_TOKEN}`;
-        log.INFO(path);
-        lastGame = await axios(path);
-    }
-    catch(err) {
+        : undefined;
+
+    const pathLastGameData = `https://${realm}.api.riotgames.com/lol/match/v4/matches/${matchId}?api_key=${config.RIOT_API_TOKEN}`;
+    const pathLastGameTimeline = `https://${realm}.api.riotgames.com/lol/match/v4/timelines/by-match/${matchId}?api_key=${config.RIOT_API_TOKEN}`;
+    const lastGameData:any = await axios(pathLastGameData).catch(err => {
+        log.WARN(err);
+        msg.channel.send(createEmbed(`❌Cannot get match data`, [{ title: '\_\_\_', content: `Fetching game ${matchId} data failed.` }]));
+        return;
+    })    
+    let lastGameTimeline:any = await axios(pathLastGameTimeline).catch(err =>{
         log.WARN(err);
-        msg.channel.send(createEmbed(`❌Cannot get match data`, [{ title: '\_\_\_', content: `Fetching match ${matchId} data failed.` }]));
+        msg.channel.send(createEmbed(`❌Cannot get match data`, [{ title: '\_\_\_', content: `Fetching timeline of game ${matchId} data failed.` }]));
         return;
+    })
+
+    let players:any = [ ];
+    let winningTeam:any = '';
+
+    lastGameData.data.teams.map((team:any) => {
+        if(team.win === 'Win')
+            winningTeam = team.teamId;
+    })
+    lastGameData.data.participantIdentities.map((participant:any) => {
+        players.push({
+            gameId: participant.participantId,
+            id: participant.player.accountId,
+            name: participant.player.summonerName,
+        })
+    })
+    lastGameData.data.participants.map((participant:any) => {
+        const i = players.findIndex((player:any) => player.gameId === participant.participantId);
+        players[i].win = participant.teamId === winningTeam;
+        players[i].teamId = participant.teamId;
+        players[i].championId = participant.championId;
+        players[i].champion = champions.find(champ => champ.id == participant.championId);
+        players[i].summ1 = participant.spell1Id;
+        players[i].summ2 = participant.spell2Id;
+        players[i].fbkill = participant.stats.firstBloodKill;
+        players[i].fbassist = participant.stats.firstBloodAssist;
+        players[i].role = participant.timeline.role;
+        players[i].lane = participant.timeline.lane;
+    })
+    const ourPlayer = players.find((player:any) => player.id === playerId);
+    const enemies = players.filter((player:any) => player.lane === ourPlayer.lane && player.teamId !== ourPlayer.teamId);
+    const lane = cache["lanes"].find((lane:any) => lane.lane == ourPlayer.lane)
```

Comment:
What is this cache exactly?

Reply:
At the bot's startup, bot loads the static data from my database into this cache. I didn't have time to document it anywhere yet. In majority of cases it's just copy of static data from Riot's API.
In this case, `lanes` associates the lane's name with the corresponding icon to be displayed in the command 
![image](https://user-images.githubusercontent.com/12732348/66752560-02bbaa80-ee92-11e9-8ab9-18f29c139fd1.png)